src/java.base/share/classes/sun/net/www/protocol/http/AuthenticationInfo.java

```diff
@@ -152,14 +152,9 @@ private static AuthenticationInfo requestAuthentication(String key, Function<Str
 
             // Otherwise, if no request is in progress, record this
             // thread as performing authentication and returns null.
-            Thread t, c;
-            c = Thread.currentThread();
-            if ((t = requests.get(key)) == null) {
-                requests.put (key, c);
-                assert cached == null;
-                return cached;
-            }
-            if (t == c) {
+            Thread c = Thread.currentThread();
+            Thread t = requests.putIfAbsent(key, c);
+            if (t == null || t == c) {
                 assert cached == null;
                 return cached;
```

Comment:
Hello Andrey, while you are in this code, I think changing these 2 lines:

```
assert cached == null;
return cached;
```
to just:

```
return null;
```
would be better. There's already a `if (cached != null) return cached;` code, a few lines above and after that line there's no other modifications to this `cached` local variable, so changing this line to just return null would remove any confusion while reading this code.

Reply:
Thank you for that change. The changes in this PR looks fine to me.